### ISSUE TITLE
Dependencies seem to use LGPL-3.0+

### ISSUE THREAD

**Issue by AdamMajer** [cockpit-project/cockpit#15433]
Cockpit version: 238.1

From the dependencies,

```
    "@patternfly/react-console": {
      "version": "1.10.46",
      "resolved": "https://registry.npmjs.org/@patternfly/react-console/-/react-console-1.10.46.tgz",
      "integrity": "sha512-uk84hxQ1OjfJwctikPX/jfYxVLHG6pqA+EH7l2BcDDEDqfm0yBqa859Klg6z2wxRt30erGy0ag4tHOQIx2MymA==",
      "requires": {
        "@novnc/novnc": "^1.0.0",
        "@spice-project/spice-html5": "^0.2.1",
```

The spice-html5 is licensed under LGPL-3.0+. Does this mean that Cockpit should be licensed under LGPL-3.0+ instead now as the spice-project is incorporated in the generated files?

**Comment by martinpitt** [comment]
Cockpit's own code is LGPL 2.1+. It contains literally hundreds of NPM modules which have all sorts of licenses, mostly MIT, various BSDs, Apache, and MPL. For Debian we [automatically generate the debian/copyright file](https://tracker.debian.org/media/packages/c/cockpit/copyright-238-1), as this is really complex. Fedora does not have an equivalent notion of such a copyright file.

So as a legal-naïve person it just seems to me that cockpit's own code must have a license that's *compatible* to all the libraries that it bundles -- it's fairly similar to what a C program does when it links to glibc, OpenSSL, kerberos, etc. LGPL-2.1+ ought to be compatible with LGPL-3+.

If you have any further or more well-founded knowledge about this legal aspect, I'm all ears of course.

I am just hesitant to change cockpit's license, as this is a ginormous, and often even impossible, ordeal -- every single copyright holder *ever* needs to agree to that, and there is ~ 180 of them.

**Comment by AdamMajer** [comment]
Yes, indeed, you are correct. Cockpit's own code must only be compatible. So there is no need to change the license of the sources. But at the same time, the aggregate distribution (compiled sources with the dist/*) should be licensed with a license that is valid for the aggregate. In this case it probably should be LGPL-3.0+, for the _compiled_ versions.

I guess the same argument applies for the distribution license of the tarballs as for the sources - it only needs to have a compatible license of the linked collection. I think it would then mean LGPL-3+ being the lowest common denominator?

**Comment by martinpitt** [comment]
Cockpit itself does not use/ship these dependencies any more -- all the runtime dependencies are MIT licensed now. However, the spice dependency moved to cockpit-machines, so transfering this issue.
